Update Jupyter examples

speth opened the pull request:
- Improve descriptions given in the index of Jupyter examples (in combination with Cantera/cantera-jupyter#33). For example, this:
![image](https://user-images.githubusercontent.com/644977/149842216-7d317fb2-3d1e-44d0-b89a-711510257d95.png)
becomes:
![image](https://user-images.githubusercontent.com/644977/149842261-c30f43f5-0ffc-4ebd-b399-57759ebb466f.png)

- Automatically link to any input files used in each example that come from the `data` directory (also reorganized by Cantera/cantera-jupyter#33). For example:
![image](https://user-images.githubusercontent.com/644977/149842374-784177a1-bc1c-4231-a22a-3c42deb14b27.png)

ischoegl reviewed (changes requested):
@speth ... thank you for this update. Just some minor comments from my side ...

I just pulled the artifacts and the main page looks great! Only thing I noticed is that one of the newer examples isn't added yet: [ck2yaml_demo](https://github.com/Cantera/cantera-jupyter/blob/main/input/ck2yaml_demo.ipynb)

Beyond, I think that locations of YAML files could also be pointed out [here](https://cantera.org/examples/jupyter/index.html), i.e.

> If you have Cantera and the Jupyter Notebook server installed on your local machine, simply download any Jupyter notebook by clicking the "Source" link at the top of each example page and you should be able to run it.

[As an aside, if I follow these instructions on Firefox/Fedora, I get to some cryptic looking output for the artifact when I click on 'source', although I didn't experience this on the actual website.]

Spot-checking, the data files are displayed correctly for `lithium_ion_battery.ipynb`, but are missing for `equations_of_state.ipynb`. On that note, the latter notebook also has some problems with LaTeX rendering that I didn't notice elsewhere.

On `interactive_path_diagram.ipynb`, the rendering of some of the headers is not pretty (likely the wrong header level). 

Some of the issues I am pointing out are obviously not website related, but should be fixed by a minor update on `cantera-jupyter`.

bryanwweber commented:
> Some of the issues I am pointing out are obviously not necessarily website related, but should be fixed by a minor update on `cantera-jupyter`.

I'm happy to add these to https://github.com/Cantera/cantera-jupyter/pull/34

speth commented:
I fixed the missing `ck2yaml` example (a new category should have been added when that example was added), added directions explaining that the data files also need to be downloaded, and fixed the cases where the data files weren't being detected.

One case that I did _not_ resolve is the input files for the `ck2yaml` example, which breaks the pattern of using the `data` directory for all input files. I think those input files should be moved to conform to the pattern of always using the `data` directory, but that needs to be dealt with on the `cantera-jupyter` repository.

I also did not see a rendering problem with `equations_of_state.ipynb` - could you be more specific (not that I think that has anything to do with this PR).

ischoegl commented:
> I fixed the missing `ck2yaml` example (a new category should have been added when that example was added) [...]

Thank you! Example was merged very recently (after creation of this PR!), although I wrote it quite some time ago. Honestly didn't think about it until it came up here ...

> One case that I did not resolve is the input files for the `ck2yaml` example, which breaks the pattern of using the data directory for all input files. I think those input files should be moved to conform to the pattern of always using the data directory, but that needs to be dealt with on the `cantera-jupyter` repository.

I think that fell between the cracks - changes date back way before Cantera/cantera-jupyter#33, although both PR's were merged the same day. I am hesitant to fix this now as it will mess with Cantera/cantera-jupyter#34. Imho, there should be some distinction as CK input is a separate category though.

> I also did not see a rendering problem with `equations_of_state.ipynb` - could you be more specific (not that I think that has anything to do with this PR).

I checked again today, and it may have something to do with my local OS. When viewing the downloaded artifacts yesterday, I had stray `$`'s in my output, but they're gone today.

ischoegl commented on themes/cantera/templates/jupyter-example-index.tmpl:
```diff
@@ -6,16 +6,21 @@
+server installed on your local machine, simply download any Jupyter notebook by clicking
+the "Source" link at the top of each example page and you should be able to run it. For
+some examples, you will also need to download the additional data files listed below the
+"Source" link. These data files should either be placed in a subdirectory named "data"
+one level up from the directory containing the notebook file, or the notebook can be
+modified to point to the location of these data files on your computer.
```
On second thought: would it make sense to just tell users to clone the `cantera-jupyter` repo?

speth replied:
No, I don't think we want to put the barrier of using Git between our users and some of our most readily accessible examples.

ischoegl replied:
Ok. In this case I’m ok with merging … :shipit:

speth replied:
Well, on second thought, I don't think it hurts to mention the option of cloning the repo as an alternative.

bryanwweber commented:
> I fixed the missing `ck2yaml` example (a new category should have been added when that example was added),

Sorry, I should have caught that at review.

> but that needs to be dealt with on the `cantera-jupyter` repository.

Will do.
